### ISSUE TITLE
Open ports on a linux box, using version 2.0.0.pre3

### ISSUE THREAD

**Issue by ddanciu** [MSOpenTech/vagrant-azure#161]
Hi,

I am not able to open ports on version 2.0.0.pre3

```
$ vagrant plugin list
vagrant-azure (2.0.0.pre3)
  - Version Constraint: 2.0.0.pre3
```

I tried using the setting:

```
  config.vm.provider :azure do |azure, override|
       azure.location = 'westeurope'
       azure.vm_image_urn='openlogic:CentOS:7.3:latest'
       azure.tcp_endpoints='8000:9000'     
  end
```

And during setup seems to be picked up:
```
$ vagrant up --provider=azure
Bringing machine 'default' up with 'azure' provider...
==> default: Launching an instance with the following settings...
==> default:  -- Management Endpoint: https://management.azure.com
==> default:  -- Subscription Id: ************
==> default:  -- Resource Group Name: ******
==> default:  -- Location: westeurope
==> default:  -- Admin Username: vagrant
==> default:  -- VM Name: *********
==> default:  -- VM Size: Standard_DS2_v2
==> default:  -- Image URN: openlogic:CentOS:7.3:latest
==> default:  -- TCP Endpoints: 8000:9000
==> default:  -- DNS Label Prefix: *******
==> default:  -- Create or Update of Resource Group: ********
==> default:  -- Starting deployment
```

I checked the code and the template but I don't see where this is handled.

Can you please point me to the correct settings or towards a workaround.

Thanks,
Dan

**Comment by devigned** [comment]
Ahh... that needs to be added right here: https://github.com/Azure/vagrant-azure/blob/v2.0/templates/arm/deployment.json.erb#L131-L176

**Comment by ddanciu** [comment]
Hi,

When do you expect that 2.0.0.pre4 is going to be available?

Thanks,
Dan

**Comment by gabidombrowski** [comment]
Also looking for this feature. The azure cli command to do that is this btw, in case that works as a workaround for you in the meantime:

az network nsg rule create --resource-group myResourceGroup \
    --nsg-name myNetworkSecurityGroup --name myNetworkSecurityGroupRule \
    --protocol tcp --direction inbound --priority 1000 \
    --source-address-prefix '*' --source-port-range '*' \
    --destination-address-prefix '*' --destination-port-range 80 --access allow

https://docs.microsoft.com/en-us/azure/virtual-machines/virtual-machines-linux-nsg-quickstart

**Comment by devigned** [comment]
Thank you @ddanciu for the pull request and feedback. This is released in 2.0.0.pre4.

@pitchdiesel check out `az vm open-port`. We were banging our heads on the verbosity of the networking commands, so we introduced a much simpler version. 

We take the VM, find the NSG it belongs to, then add the rule. Seems much more natural than providing the NSG for most VM use cases.

```
$ az vm open-port -h

Command
    az vm open-port: Opens a VM to inbound traffic on specified ports by adding a security rule to
    the network security group (NSG) that is attached to the VM's network interface (NIC) or subnet.

Arguments
    --port   [Required]: The port or port range (ex: 80-100) to open inbound traffic to. Use '*' to
                         allow traffic to all ports.
    --apply-to-subnet  : Allow inbound traffic on the subnet instead of the NIC.
    --nsg-name         : The name of the network security group to create if one does not exist.
                         Ignored if an NSG already exists.
    --priority         : Rule priority, between 100 (highest priority) and 4096 (lowest priority).
                         Must be unique for each rule in the collection.  Default: 900.

Resource Id Arguments
    --ids              : One or more resource IDs (space delimited). If provided, no other 'Resource
                         Id' arguments should be specified.
    --name -n          : The name of the virtual machine to open inbound traffic on.
    --resource-group -g: Name of resource group.

Global Arguments
    --debug            : Increase logging verbosity to show all debug logs.
    --help -h          : Show this help message and exit.
    --output -o        : Output format.  Allowed values: json, jsonc, table, tsv.  Default: json.
    --query            : JMESPath query string. See http://jmespath.org/ for more information and
                         examples.
    --verbose          : Increase logging verbosity. Use --debug for full debug logs.

Examples
    Open all ports on a VM to inbound traffic.
        az vm open-port -g MyResourceGroup -n MyVm --port *
    Open a range of ports on a VM to inbound traffic with the highest priority.
        az vm open-port -g MyResourceGroup -n MyVm --port 80-100 --priority 100
    Open all ports for multiple VMs by Ids
        az vm open-port --ids $(az vm list -g MyResourceGroup --query "[].id" -o tsv)
```

**Comment by devigned** [comment]
I'm closing as this has been released.
